cloudinit/net/dhcp.py

```diff
@@ -198,82 +170,93 @@ def start_service(cls, dhcp_interface: str, distro):
     def stop_service(cls, dhcp_interface: str, distro):
         distro.manage_service("stop", cls.client_name, rcs=[0, 1])
 
+    @staticmethod
+    def parse_static_routes(routes: str) -> List[Tuple[str, str]]:
+        return []
+
+    def get_newest_lease(self, interface: str) -> Dict[str, Any]:
```

Comment:
All subclasses implement this method any reason to avoid just making it `@abc.abstractmethod` ?

Reply:
Since we want this method to be implemented by all subclasses we should drop the `return {}` below correct? Otherwise subclasses won't actually have to implement it.
```diff
--- a/cloudinit/net/dhcp.py
+++ b/cloudinit/net/dhcp.py
@@ -176,7 +176,7 @@ class DhcpClient(abc.ABC):
 
     @abc.abstractmethod
     def get_newest_lease(self, interface: str) -> Dict[str, Any]:
-        return {}
+        """Get the most recent lease from the ephemeral phase as a dict."""
 
 
 class IscDhclient(DhcpClient):
```